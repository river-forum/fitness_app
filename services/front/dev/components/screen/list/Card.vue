<template>
  <div>
    <div class="flex items-center space-x-4">
      <p
        v-for="(i, n) in NarrowDown"
        :key="i"
        class="cursor-pointer bg-green-500 hover:opacity-75 px-3 py-2 text-white"
        @click="narrowDownClickListener(n)"
      >
        {{ n }}
      </p>
    </div>

    <div class="flex items-center flex-wrap space-x-3">
      <p v-for="(d, i) in computedDates" :key="i">
        {{ d }}
      </p>
    </div>
    {{ currentNarrowDown }}
  </div>
</template>
<script>
import { defineComponent, ref, computed } from '@nuxtjs/composition-api'

const NarrowDown = {
  DAY: 'day',
  WEEK: 'week',
  MONTH: 'month',
}

export default defineComponent({
  props: {
    dates: {
      type: Array,
      required: true,
    },
  },
  setup({ dates }) {
    const currentNarrowDown = ref(NarrowDown.DAY)

    const computedDates = computed(() => {
      switch (currentNarrowDown.value) {
        case NarrowDown.DAY:
          return dates.map((date) => date.getDay())
        case NarrowDown.WEEK:
          return dates.map((date) => date.getFullYear())
        case NarrowDown.MONTH:
          return dates.map((date) => date.getMonth())
      }

      throw new Error('不正な日付フォーマットの選択肢です。')
    })

    function narrowDownClickListener(val) {
      currentNarrowDown.value = val.toLowerCase()
    }
    return {
      NarrowDown,
      computedDates,
      currentNarrowDown,
      narrowDownClickListener,
    }
  },
})
</script>
