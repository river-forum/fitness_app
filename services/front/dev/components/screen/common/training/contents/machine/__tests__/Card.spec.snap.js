import { mount } from '@vue/test-utils'
import Card from '../Card.vue'

it('Card Component renders correctly', () => {
  const app = mount(Card)

  expect(app.html()).toMatchSnapshot()
})
