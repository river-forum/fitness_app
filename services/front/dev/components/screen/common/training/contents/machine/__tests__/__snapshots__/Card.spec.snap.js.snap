// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
"<div class=\\"z-50 bg-white machine-card\\"><button class=\\"absolute z-50 w-12 h-12 bg-gray-600 rounded-full close\\"><span class=\\"absolute top-0 bottom-0 left-0 right-0 w-3/5 h-px m-auto bg-black close__line\\"></span> <span class=\\"absolute top-0 bottom-0 left-0 right-0 w-3/5 h-px m-auto bg-black close__line\\"></span></button>
  <figure class=\\"relative overflow-hidden machine-card__img-wrap\\"><img src=\\"https://placehold.jp/730x390.png\\" alt=\\"\\" class=\\"absolute top-0 bottom-0 left-0 right-0 object-cover w-full h-full m-auto machine-card__img\\"></figure>
  <div class=\\"flex items-center text-white machine-card__label\\">
    <figure class=\\"flex items-center justify-center py-8 machine-card__icon\\"><img src=\\"assets/icons/dumbbell.svg\\" alt=\\"\\" width=\\"25rem\\" height=\\"25rem\\"></figure>
    <p class=\\"py-8 pl-4 machine-card__title\\">

    </p>
  </div>
  <div class=\\"mt-6 machine-card__body\\">
    <div class=\\"h-48 px-8 overflow-auto l-container\\">
      <div class=\\"inner-wrap\\">
        <inputdate></inputdate>
        <inputnumber type=\\"weight\\" class=\\"mt-6\\"></inputnumber>
        <inputnumber type=\\"times\\" class=\\"mt-6\\"></inputnumber>
        <inputnumber type=\\"set\\" class=\\"mt-6\\"></inputnumber> <button button-inner-text=\\"記録する\\" class=\\"mx-auto mt-16\\"></button>
      </div>
    </div>
  </div>
</div>"
`;
